File: nhglib-core/src/io/github/voidzombie/nhglib/utils/files/JsonLoader.java
package io.github.voidzombie.nhglib.utils.files;

import com.badlogic.gdx.assets.AssetDescriptor;
import com.badlogic.gdx.assets.AssetLoaderParameters;
import com.badlogic.gdx.assets.AssetManager;
import com.badlogic.gdx.assets.loaders.AsynchronousAssetLoader;
import com.badlogic.gdx.assets.loaders.FileHandleResolver;
import com.badlogic.gdx.files.FileHandle;
import com.badlogic.gdx.utils.Array;
import com.badlogic.gdx.utils.JsonReader;
import com.badlogic.gdx.utils.JsonValue;

/**
 * Created by Fausto Napoli on 19/12/2016.
 */
public class JsonLoader extends AsynchronousAssetLoader<JsonValue, JsonLoader.JsonParameter> {
    public JsonLoader(FileHandleResolver resolver) {
        super(resolver);
    }

    @Override
    public void loadAsync(AssetManager manager, String fileName, FileHandle file, JsonParameter parameter) {
    }

    @Override
    public JsonValue loadSync(AssetManager manager, String fileName, FileHandle file, JsonParameter parameter) {
        return new JsonReader().parse(file);
    }

    @Override
    public Array<AssetDescriptor> getDependencies(String fileName, FileHandle file, JsonParameter parameter) {
        return null;
    }

    public static class JsonParameter extends AssetLoaderParameters<JsonValue> {
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/assets/loaders/JsonLoader.kt
package io.github.movementspeed.nhglib.assets.loaders

import com.badlogic.gdx.assets.AssetDescriptor
import com.badlogic.gdx.assets.AssetLoaderParameters
import com.badlogic.gdx.assets.AssetManager
import com.badlogic.gdx.assets.loaders.AsynchronousAssetLoader
import com.badlogic.gdx.assets.loaders.FileHandleResolver
import com.badlogic.gdx.files.FileHandle
import com.badlogic.gdx.utils.Array
import com.badlogic.gdx.utils.JsonReader
import com.badlogic.gdx.utils.JsonValue

/**
 * Created by Fausto Napoli on 19/12/2016.
 */
class JsonLoader(resolver: FileHandleResolver) : AsynchronousAssetLoader<JsonValue, JsonLoader.JsonParameter>(resolver) {

    override fun loadAsync(manager: AssetManager, fileName: String, file: FileHandle, parameter: JsonParameter) {}

    override fun loadSync(manager: AssetManager, fileName: String, file: FileHandle, parameter: JsonParameter): JsonValue {
        return JsonReader().parse(file)
    }

    override fun getDependencies(fileName: String, file: FileHandle, parameter: JsonParameter): Array<AssetDescriptor<*>>? {
        return null
    }

    class JsonParameter : AssetLoaderParameters<JsonValue>()
}
